### PR TITLE
Move Metadata API client to Python SDK

### PR DESCRIPTION
**Pull request by Jesus89**
- [x] Refactor RepoClient
- [x] Add unit tests

### CODE REVIEW

**Comment by rafatower** [comment]
Related to https://github.com/CartoDB/carto-python/pull/163

File: cartoframes/data/observatory/catalog/repository/repo_client.py [hunk 1]
```diff
@@ -1,125 +1,83 @@
...
 class RepoClient:
 
-    __instance = None
-
     def __init__(self):
-        self._user_credentials = None
+        self._do_dataset = None
+        default_credentials = Credentials(DEFAULT_USER)
+        default_auth_client = default_credentials.get_api_key_auth_client()
+        self._defautl_do_dataset = DODataset(auth_client=default_auth_client)
```
Comment (rafatower) [inline cartoframes/data/observatory/catalog/repository/repo_client.py]: Minor typo: `s/_defautl_do_dataset/_default_do_dataset/` (consistently repeated below)

Comment (rafatower) [reply cartoframes/data/observatory/catalog/repository/repo_client.py]: Aside and out of scope: not sure those default credentials `do-metadata` with `default_public` api key work with the DO API, but they should, and we should test it (if they don't, it shall be just a matter of removing a line in do api)

File: tests/unit/data/observatory/catalog/repository/test_repo_client.py [hunk 2]
```diff
@@ -1,50 +1,146 @@
...
-from cartoframes.data.observatory.catalog.repository.constants import CATEGORY_FILTER, COUNTRY_FILTER
+from unittest.mock import patch
+
 from cartoframes.data.observatory.catalog.repository.repo_client import RepoClient
-from ..mocks import mocked_do_api_requests_get_datasets
-from ..examples import db_dataset1, db_dataset2
 
 
-class TestRepoClient(object):
+class TestRepoClient:
```
Comment (rafatower) [inline tests/unit/data/observatory/catalog/repository/test_repo_client.py]: nice tests! :tada:

File: cartoframes/data/observatory/catalog/repository/repo_client.py [hunk 3]
```diff
@@ -1,125 +1,83 @@
...
-            scheme=url.scheme, netloc=url.netloc, path=path, query=query, fragment=url.fragment
-        ))
-
-    def _get_user_credentials(self):
-        return self._user_credentials or get_default_credentials() or Credentials(DEFAULT_USER)
+    def _fetch_entity(self, entity, filters=None):
+        if self._do_dataset:
+            return self._do_dataset.metadata(entity, filters)
+        else:
+            return self._defautl_do_dataset.metadata(entity, filters)
```
Comment (oleurud) [inline cartoframes/data/observatory/catalog/repository/repo_client.py]:
```suggestion
            return self._default_do_dataset.metadata(entity, filters)
```

File: cartoframes/data/observatory/catalog/repository/repo_client.py (hunk 1 above)
Comment (Jesus89) [reply cartoframes/data/observatory/catalog/repository/repo_client.py]: I tested it and it works :)

**Pull request merged**